addons/sale/models/sale.py

```diff
@@ -1390,10 +1390,9 @@ def _compute_untaxed_amount_to_invoice(self):
                 # amount and not zero. Since we compute untaxed amount, we can use directly the price
                 # reduce (to include discount) without using `compute_all()` method on taxes.
                 price_subtotal = 0.0
-                if line.product_id.invoice_policy == 'delivery':
-                    price_subtotal = line.price_reduce * line.qty_delivered
-                else:
-                    price_subtotal = line.price_reduce * line.product_uom_qty
+                umo_qty_to_consider = line.qty_delivered if line.product_id.invoice_policy == 'delivery' else line.product_uom_qty
+                price_reduce = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
+                price_subtotal = line.currency_id.round(price_reduce * umo_qty_to_consider)
```

Comment:
Is this rounding even necessary ? As I see it, the automatic rounding of `untaxed_amount_to_invoice` as monetary field should give the same result, without the intermediary rounding :thinking:.

```suggestion
                price_subtotal = price_reduce * uom_qty_to_consider
```

Reply:
Indeed :ok_hand: